pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -155,3 +165,152 @@ def check_slice(self, frame_gray):
             return False
         else:
             return None
+
+
+class Video:
+    def __init__(self, path: str) -> None:
+        self.path = path
+        self.ts = None
+
+    def check_container(self):
+        try:
+            cont = av.open(self.path)
+            n = cont.decode(video=0)
+            _ = next(n)
+        except av.AVError:
+            return
+        else:
+            return av.open(self.path)
+
+    def load_container(self):
+        return av.open(self.path)
+
+    def load_ts(self):
+        self.ts = np.load(self.ts_loc)
+
+    @property
+    def name(self) -> str:
+        file_ = os.path.split(self.path)[1]
+        return os.path.splitext(file_)[0]
+
+    @property
+    def ts_loc(self) -> str:
+        return os.path.join(self.base, f"{self.name}_timestamps.npy")
+
+    @property
+    def base(self) -> str:
+        return os.path.split(self.path)[0]
+
+    @property
+    def timestamps(self) -> np.ndarray:
+        if self.ts is None:
+            self.load_ts()
+        return self.ts
+
+
+class VideoSet:
+    def __init__(self, rec: str, name: str, fill_gaps: bool):
+        self.rec = rec
+        self.name = name
+        self.fill_gaps = fill_gaps
+        self.video_exts = ("mp4", "mjpeg", "h264")
```

Comment:
```suggestion
        self.video_exts = ("mp4", "mjpeg", "h264", "mkv", "avi")
```
We need to add some more legacy ext. They do not require specific tests.